pulsequantum/livestream/sweepsettings.py

```diff
@@ -44,7 +44,11 @@ def __init__(self, video, awg=None):
         self.col = Row(Column(self.settings, self.get_button, self.set_button), Column(self.figpane, self.upload_button, self.run_button))
 
     def config_event(self, event):
+        self.set_button.button_type = 'warning'
+        self.set_button.disabled = True
         self.config()
+        self.set_button.disabled = False
```

Comment:
What happens if self.config() fails? 
will the button stay disabled

Reply:
Yes currently it does, I have figured a way to resolve that issue though, but it still partially runs the self.config() untill the point where it fails. I'll commit the version with that resolved now

Reply:
I don't know if it is fine that the self.config() runs partially only, since it then only sets some variables and not all. But i have implemented a self.get_settings() to show the user what the current parameters are after the self.config() fails